FlexMeasures/flexmeasures
# feat: add chart action to export data to CSV

## Description

Append a chart action in the ellipses menu of each vega-lite chart, for exporting the sensor data to CSV.

## Look & Feel

![image](https://github.com/FlexMeasures/flexmeasures/assets/30658763/6b2d0ce7-e5e4-4fe8-8398-74ae15225849)

![image](https://github.com/FlexMeasures/flexmeasures/assets/30658763/89fc6615-4a48-44f2-8b95-1979e646c85a)

## To test

- Check out the `export-to-csv` branch.
- `flexmeasures run`
- Navigate to an asset page, for example: `http://localhost:5000/assets/1`
- Click on the ellipses menu (on the top right of the chart) and download the CSV
- Do the same for a sensor page, for example: `http://localhost:5000/sensors/1`

## Related Items

Adapted from https://github.com/vega/vega-embed/issues/156#issuecomment-583175096.
